analyzer/runtime/evm/client.go

```diff
@@ -313,21 +313,30 @@ func EVMMaybeUnmarshalEncryptedData(data []byte, result *[]byte) (*EVMEncryptedD
 		if err := cbor.Unmarshal(*result, &callResult); err != nil {
 			return nil, fmt.Errorf("unmarshal outer call result: %w", err)
 		}
-		if callResult.IsUnknown() {
-			switch call.Format {
-			case sdkTypes.CallFormatEncryptedX25519DeoxysII:
-				var resultEnvelope sdkTypes.ResultEnvelopeX25519DeoxysII
+		switch call.Format {
+		case sdkTypes.CallFormatEncryptedX25519DeoxysII:
+			var resultEnvelope sdkTypes.ResultEnvelopeX25519DeoxysII
+			switch {
+			// The result is encrypted.
+			case callResult.IsUnknown():
 				if err := cbor.Unmarshal(callResult.Unknown, &resultEnvelope); err != nil {
-					return nil, fmt.Errorf("outer call result unmarshal unknown: %w", err)
+					return nil, fmt.Errorf("outer call result unmarshal ok: %w", err)
```

Comment:
why is this changed to 'ok'? we're still unmarshalling the .Unknown

Reply:
fixed!